yarn-project/stdlib/src/rollup/checkpoint_root_rollup.ts

```diff
@@ -0,0 +1,216 @@
+    public previousRollups: [RollupProofData<BlockRollupPublicInputs>, RollupProofData<BlockRollupPublicInputs>],
+    public hints: CheckpointRootRollupHints,
+  ) {}
+
+  static from(fields: FieldsOf<CheckpointRootRollupPrivateInputs>) {
+    return new CheckpointRootRollupPrivateInputs(...CheckpointRootRollupPrivateInputs.getFields(fields));
+  }
+
+  static getFields(fields: FieldsOf<CheckpointRootRollupPrivateInputs>) {
+    return [fields.previousRollups, fields.hints] as const;
+  }
+
+  toBuffer() {
+    return serializeToBuffer(...CheckpointRootRollupPrivateInputs.getFields(this));
+  }
+
+  static fromBuffer(buffer: Buffer | BufferReader) {
+    const reader = BufferReader.asReader(buffer);
+    return new CheckpointRootRollupPrivateInputs(
+      [ProofData.fromBuffer(reader, BlockRollupPublicInputs), ProofData.fromBuffer(reader, BlockRollupPublicInputs)],
+      CheckpointRootRollupHints.fromBuffer(reader),
+    );
+  }
+
+  toString() {
+    return bufferToHex(this.toBuffer());
+  }
+
+  static fromString(str: string) {
+    return CheckpointRootRollupPrivateInputs.fromBuffer(hexToBuffer(str));
+  }
+
+  toJSON() {
+    return this.toBuffer();
+  }
+
+  static get schema() {
+    return bufferSchemaFor(CheckpointRootRollupPrivateInputs);
+  }
+}
+
+export class CheckpointRootSingleBlockRollupPrivateInputs {
+  constructor(
+    public previousRollup: RollupProofData<BlockRollupPublicInputs>,
+    public hints: CheckpointRootRollupHints,
+  ) {}
+
+  static from(fields: FieldsOf<CheckpointRootSingleBlockRollupPrivateInputs>) {
+    return new CheckpointRootSingleBlockRollupPrivateInputs(
+      ...CheckpointRootSingleBlockRollupPrivateInputs.getFields(fields),
+    );
+  }
+
+  static getFields(fields: FieldsOf<CheckpointRootSingleBlockRollupPrivateInputs>) {
+    return [fields.previousRollup, fields.hints] as const;
+  }
+
+  toBuffer() {
+    return serializeToBuffer(...CheckpointRootSingleBlockRollupPrivateInputs.getFields(this));
+  }
+
+  static fromBuffer(buffer: Buffer | BufferReader) {
+    const reader = BufferReader.asReader(buffer);
+    return new CheckpointRootSingleBlockRollupPrivateInputs(
+      ProofData.fromBuffer(reader, BlockRollupPublicInputs),
+      CheckpointRootRollupHints.fromBuffer(reader),
+    );
+  }
+
+  toString() {
+    return bufferToHex(this.toBuffer());
+  }
+
+  static fromString(str: string) {
+    return CheckpointRootSingleBlockRollupPrivateInputs.fromBuffer(hexToBuffer(str));
+  }
+
+  toJSON() {
+    return this.toBuffer();
+  }
+
+  static get schema() {
+    return bufferSchemaFor(CheckpointRootSingleBlockRollupPrivateInputs);
+  }
+}
+
+// Checkpoint padding circuit does not have any private inputs.
+export class CheckpointPaddingRollupPrivateInputs {
+  constructor() {}
+
+  toBuffer() {
+    return Buffer.alloc(0);
+  }
+
+  static fromBuffer(_buffer: Buffer | BufferReader) {
+    return new CheckpointPaddingRollupPrivateInputs();
+  }
+
+  toString() {
+    return '';
+  }
```

Comment:
```suggestion
  toString(): `0x${string}` {
    return '0x';
  }
```
Just for consistency